library/ssl_msg.c

```diff
@@ -3106,32 +3106,127 @@ static size_t ssl_get_reassembly_buffer_size( size_t msg_len,
 
 #endif /* MBEDTLS_SSL_PROTO_DTLS */
 
+#if defined(MBEDTLS_SSL_HS_DEFRAG_CLT)
+#define TRY_READ_FLAG_COMMIT 1
+static int ssl_defrag_try_read( mbedtls_ssl_context *ssl, size_t desired, unsigned int flags )
```

Comment:
This essentially exposes a stream read-interface to handshake data, and the function name should reflect that. How about `ssl_read_handshake_data()`?